ando/tools/generator/nwb2bidsgenerator.py

```diff
@@ -0,0 +1,214 @@
+import json
+from collections import defaultdict
+from pathlib import Path
+
+import pandas as pd
+from pynwb import NWBHDF5IO
+from pynwb.ecephys import ElectricalSeries
+
+from ando.AnDOChecker import is_valid
+
+
+def bep_organize(dataset_path, output_path=None, move_nwb=False,
+                 re_write=True, validate=True, **kwargs):
+    """
+    organize data according to teh BIDS extension proposal
+    Parameters
+    ----------
+    re_write: bool
+        if true, will rewrite an already existing output path
+    move_nwb: bool
+        if true, will move the nwb file in the new directory structure,
+        else will create a symlink
+    output_path: [str, Path]
+        parent path to the BIDS folder structure
+    dataset_path : [str, Path]
+        path to the folder containing all the nwb datasets that need organization.
+    validate: bool
+        to validate the dataset using the ANdoChecker
+    kwargs:
+        probe_type: acute/chronic
+    """
+    dataset_path = Path(dataset_path)
+    if output_path is None:
+        output_path = dataset_path.parent/'BIDSExt'/dataset_path.name
+    if re_write and output_path.exists():
+        output_path.unlink()
+    participants_df = pd.DataFrame(
+        columns=['Species', 'ParticipantID', 'Sex', 'Birthdate', 'Age', 'Genotype', 'Weight'])
+    dataset_desc_json = None
+
+    dataset_path = Path(dataset_path)
+    file_count = 0
+    sessions_count = 0
+    sub_ses_dict = defaultdict(list)
+    for nwb_file in dataset_path.glob('**/*.nwb'):
+        file_count += 1
+        channels_df = pd.DataFrame(
+            columns=['channel_id', 'Contact_id', 'type', 'units', 'sampling_frequency',
+                     'unit_conversion_multiplier'])
+        contacts_df = pd.DataFrame(
+            columns=[
+                'x',
+                'y',
+                'z',
+                'impedance',
+                'contact_id',
+                'probe_id',
+                'Location'])
+        probes_df = pd.DataFrame(columns=['probeID', 'type'])
+
+        with NWBHDF5IO(str(nwb_file), 'r') as io:
+            nwbfile = io.read()
+
+            # subject info:
+            if nwbfile.subject is not None:
+                sb = nwbfile.subject
+                if sb.subject_id is not None:
+                    subject_label = f'sub-{sb.subject_id}'
```

Comment:
Here it would be important to ensure that the subject label is consistent with the BIDS specs in the sense that the <label> part of `sub-<label>` is supposed to be alphanumeric and can not contain any other characters, e.g. `-` or `_`. Or is this restriction also already implemented in the NWB specifications? Then we could just rely on that...

Reply:
fixed https://github.com/INT-NIT/AnDO/pull/73/commits/43436e3a0c3f9a329fd0461953debce226aeee24